ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -1108,13 +1107,7 @@ knownIntersectionStateTop cfgEnv dynEnv intEnv =
         recordHeaderArrival
       } = headerInFutureCheck
 
-    -- Request the next message (roll forward or backward), unless our chain
```

Comment:
What do you think about adding a comment on the MsgRollForward handler explaining why it's sufficient for it to be the only place that checks for the intersection?

Reply:
Added a comment to `rollForward` :+1: